src/inmanta/module.py

```diff
@@ -2342,3 +2572,92 @@ def get_all_python_requirements_as_list(self) -> List[str]:
+            return False
+        pkg_name: str = ModuleV2.get_package_name_for(module_name)
+        return any(
+            r
+            for r in config_parser.get("options", "install_requires").split("\n")
+            if r and Requirement.parse(r).key == pkg_name
+        )
+
+
+class YamlMetadataFileWithRequiresField:
+    @classmethod
+    def add_module_requirement_and_write(
+        cls,
+        proj_or_v1_mod: Union[Project, ModuleV1],
+        requirement: InmantaModuleRequirement,
+        ignore_version_constraint: bool = False,
+    ) -> None:
+        """
+        Updates the metadata file of the given project or module by adding the given requirement the `requires` section.
+
+        :param proj_or_v1_mod: Project or module that should receive the new requirement.
+        :param requirement: The requirement to add.
+        :param ignore_version_constraint: If True, the version constraint in `requirement` will be left out of the .yml file.
+        """
+        # Parse cfg file
+        content: CommentedMap = PreservativeYamlParser.parse(proj_or_v1_mod.get_metadata_file_path())
+        line_to_add = str(requirement) if not ignore_version_constraint else requirement.key
+        # Update requires
+        if "requires" in content:
+            existing_matching_reqs: List[str] = [
+                r for r in content["requires"] if InmantaModuleRequirement.parse(r).key == requirement.key
+            ]
+            for r in existing_matching_reqs:
+                content["requires"].remove(r)
+            content["requires"].append(line_to_add)
+        else:
+            content["requires"] = [line_to_add]
+        # Write file back to disk
+        PreservativeYamlParser.dump(proj_or_v1_mod.get_metadata_file_path(), content)
+
+    @classmethod
+    def has_module_requirement(cls, proj_or_v1_mod: Union[Project, ModuleV1], module_name: str) -> bool:
+        """
+        Returns true iff the given module is present in the `requires` list of the given project or module metadata file.
+        """
+        content: CommentedMap = PreservativeYamlParser.parse(proj_or_v1_mod.get_metadata_file_path())
+        if "requires" not in content:
+            return False
+        return any(r for r in content["requires"] if InmantaModuleRequirement.parse(r).key == module_name)
+
+    @classmethod
+    def remove_module_requirement_and_write(cls, proj_or_v1_mod: Union[Project, ModuleV1], module_name: str) -> None:
+        """
+        Updates the metadata file of the given project or module by removing the given requirement frm the `requires` section.
+        """
+        if not cls.has_module_requirement(proj_or_v1_mod, module_name):
+            return
+        content: CommentedMap = PreservativeYamlParser.parse(proj_or_v1_mod.get_metadata_file_path())
+        content["requires"] = [r for r in content["requires"] if InmantaModuleRequirement.parse(r).key != module_name]
```

Comment:
I think this needs a check `"requires" in content` like the other methods have.

Reply:
This is already covered by the `if not self.has_module_requirement(module_name)` check